src/main/java/libraryapp/dao/impl/AuthorDaoImpl.java

```diff
@@ -0,0 +1,49 @@
+package libraryapp.dao.impl;
+
+import libraryapp.dao.AuthorDao;
+import libraryapp.entity.Author;
+
+import java.util.List;
+import javax.persistence.TypedQuery;
+import org.hibernate.SessionFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class AuthorDaoImpl implements AuthorDao {
+
+    @Autowired
+    private SessionFactory sessionFactory;
+
+    @Override
+    public void add(Author author) {
+        sessionFactory.getCurrentSession().save(author);
+    }
+
+    @Override
+    public List<Author> getAll() {
+        @SuppressWarnings("unchecked")
```

Comment:
try removing SuppressWarnings in all methods
Bohdan write how to do it in slack

Reply:
oke